server/resolvers/article/queries.js

```diff
@@ -2,23 +2,38 @@ import { AVERAGE_WORDS_PER_MINUTE } from '../../utils/constants'
 import { databaseApi, searchApi } from '../../endpoints'
 
 const articleQueries = {
-  articles: async (_, { search }) => {
+  articles: async (_, { search }, { cache }) => {
     const payload = await searchApi(search)
 
     if (payload.error) return new Error(JSON.stringify(payload.error))
 
     const articleIds = [...payload.data.articles]
     const articles = []
+    const requests = []
+
+    for (const id of articleIds) {
+      if (await cache.exists(id)) {
```

Comment:
What happens if the id in the cache exists but the road regulations document has been updated. Should we drop the cache at some point?

Reply:
Yes, that would be a nice stretch for the parser service.

Whenever it re-parses the document in-memory, it should drop the cache content so it can start fresh and store the updated data.